pkg/tgc/tgc.go

```diff
@@ -415,18 +417,24 @@ func getAvailableNetBatPairings(namespace, podName, pairingStr string) ([]util.B
 			continue
 		}
 		if source.Net != "" {
+			ipFound := false
 			if srcIface, ok := netIfNameMap[namespace+"/"+source.Net]; ok {
 				source.Interface = srcIface
 				if srcIfaceIPAddress, ok := ifNameAddressMap[srcIface]; ok {
+					ipFound = true
 					source.IP = srcIfaceIPAddress
 				}
 			}
+			if !ipFound {
+				logrus.Errorf("no interface present for given source network %s", source.Net)
+			}
 		} else if source.Interface != "" {
 			if srcIfaceIPAddress, ok := ifNameAddressMap[source.Interface]; ok {
 				source.IP = srcIfaceIPAddress
+			} else {
+				logrus.Errorf("no interface present for given source interface name %s", source.Interface)
 			}
-		}
-		if source.IP == "" {
+		} else if source.IP == "" {
```

Comment:
Actually, the only legal case if neither net nor interface is present in source is that the source interface is the primary eth0 interface. Checking for source.IP=="" is redundant as the IP is never included in the sourceStr. It is only populated by TGC.

Reply:
@JanScheurich, yes, good find! removed` source.IP==""` check now.